src/api/category.ts

```diff
@@ -0,0 +1,70 @@
+import { QueryClientConfig, UUID} from '../types';
+import { buildGetCategoriesRoute, buildGetCategoryInfoRoute, buildGetItemCategoryRoute, 
+  buildGetItemsInCategoryRoute, buildPostItemCategoryRoute, buildDeleteItemCategoryRoute, GET_CATEGORY_TYPES_ROUTE } from './routes';
+import { DEFAULT_DELETE, DEFAULT_GET, DEFAULT_POST, failOnError } from './utils';
+
+export const getCategoryTypes = async ({ API_HOST }: QueryClientConfig) => {
+  const res = await fetch(`${API_HOST}/${GET_CATEGORY_TYPES_ROUTE}`, DEFAULT_GET).then(
+    failOnError,
+  );
+
+  return res.json();
+};
+
+export const getCategories = async ({ API_HOST }: QueryClientConfig, typeId?: UUID[], ) => {
+  console.log(typeId);
+  console.log(buildGetCategoriesRoute(typeId));
+  const res = await fetch(`${API_HOST}/${buildGetCategoriesRoute(typeId)}`, DEFAULT_GET).then(
+    failOnError,
+  );
+
+  return res.json();
+};
+
+export const getCategoryInfo = async (categoryId: UUID, { API_HOST }: QueryClientConfig) => {
+    const res = await fetch(`${API_HOST}/${buildGetCategoryInfoRoute(categoryId)}`, DEFAULT_GET).then(
+      failOnError,
+    );
+  
+    return res.json();
+  };
+
+export const getItemCategory = async (itemId: UUID, { API_HOST }: QueryClientConfig) => {
+    const res = await fetch(`${API_HOST}/${buildGetItemCategoryRoute(itemId)}`, DEFAULT_GET).then(
```

Comment:
Are you fetching multiple item categories? If yes, use plural `getItemCategories`, `buildGetItemCategoriesRoute`.